packages/react-component-library/src/components/Badge/Badge.tsx

```diff
@@ -1,6 +1,7 @@
 import React from 'react'
 import classNames from 'classnames'
 
+import { PropsWithClassName } from '../../types/PropsWithClassName'
```

Comment:
I think if you name this file `PropsWithClassName.d.ts` you can get away without the explicit import at the top of each file. The `d` in the filename makes it a typescript 'ambient' module.

Reply:
It appears to work but only because I added `PropsWithClassName` to `Link.d.ts` :face_with_head_bandage: As soon as the imports are removed, `PropsWithClassName` is omitted from the bundled types again.

I was under the impression that declaration files are mainly for defining types for e.g. a third-party JS library without native types.